### PR TITLE
Replace ObjectSelector with Selector

### LINKED ISSUE #493: Selectors should inherit from Selector, not ObjectSelector
ObjectSelector is old, deprecated API, replaced with Selector. However, FileSelector, ListSelector, and MultiFileSelector still inherit from ObjectSelector, making it difficult to remove and making those classes confusing to work with.  Swapping ObjectSelector and Selector in the class hierarchy shouldn't affect anything not testing those exact classes, and we should do that soon to avoid future confusion.

### PR DESCRIPTION
**Pull request by jbednar**
Fixes #493

### CODE REVIEW

**Comment by jbednar** [comment]
Note that ListSelector, FileSelector, and MultiFileSelector all still use the convention from ObjectSelector where the first argument is the default value, not the objects list, and where the default value must be specified since it won't be computed from the objects list. Changing those things breaks backwards compatibility, so shouldn't be done lightly.

**Comment by jbednar** [comment]
Added a fix for https://github.com/holoviz/param/issues/307 .

File: param/__init__.py [hunk 1]
```diff
@@ -1149,10 +1149,14 @@ def get_range(self):
         raise NotImplementedError("get_range() must be implemented in subclasses.")
 
 
-class ObjectSelector(SelectorBase):
+class Selector(SelectorBase):
     """
     Parameter whose value must be one object from a list of possible objects.
 
+    By default, if no default is specfied, picks the first object from
```
Comment (jlstevens) [inline param/__init__.py:1156]: Typo: `specfied -> specified`. I'll make a new PR with these minor fixes now this has been merged.

File: param/__init__.py [hunk 2]
```diff
@@ -1174,11 +1178,27 @@ class ObjectSelector(SelectorBase):
 
     __slots__ = ['objects', 'compute_default_fn', 'check_on_set', 'names']
 
-    # ObjectSelector is usually used to allow selection from a list of
+    # Selector is usually used to allow selection from a list of
     # existing objects, therefore instantiate is False by default.
-    def __init__(self, default=None, objects=None,  instantiate=False,
+    def __init__(self, objects=None, default=None, instantiate=False,
```
Comment (jlstevens) [inline param/__init__.py:1183]: This order is indeed better but in theory could break anyone passing in the arguments by position. This causing an issue is very unlikely though as I don't think I've ever seen that style of signature so overall this is better.

**Comment by jlstevens** [comment]
Looks good! 

Sorry for not reviewing before merge but I don't see anything I would have objected to (other than a minor typo fix that I'll push a PR for now).

File: param/__init__.py (hunk 1 above)
Comment (jlstevens) [reply param/__init__.py:1156]: Fixed in #507

**Pull request merged**